pkg/services/serviceaccounts/extsvcaccounts/service.go

```diff
@@ -380,6 +380,56 @@ func (esa *ExtSvcAccountsService) getExtSvcAccountToken(ctx context.Context, org
 	return newKeyInfo.ClientSecret, nil
 }
 
+func genTokenWithRetries(ctxLogger log.Logger, extSvcSlug string) (satokengen.KeyGenResult, error) {
+	var newKeyInfo satokengen.KeyGenResult
+	var err error
+	var ok bool
+	retry := 0
+	for !ok && retry < maxTokenGenRetries {
+		newKeyInfo, err = satokengen.New(extSvcSlug)
+		if err != nil {
+			return satokengen.KeyGenResult{}, err
+		}
+
+		if !strings.Contains(newKeyInfo.ClientSecret, "\x00") {
+			ok = true
+			continue
```

Comment:
I think you can do `return newKeyInfo, nil` here, and after the `for` loop just return the error. This way you can drop the `ok` var too.

Reply:
Good catch. I extracted the code from the calling function where I could not return.